src/include/cell.i

```diff
@@ -924,6 +908,43 @@ __wt_cell_unpack_dsk(
     }
 
     WT_IGNORE_RET(__wt_cell_unpack_safe(session, dsk, cell, unpack, NULL));
+
+    /*
```

Comment:
This is the main guts of my change. This logic corresponds to the table that I showed in the knowledge sharing session this week.

Reply:
Done.